src/main/java/com/myhome/services/springdatajpa/PaymentSDJpaService.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 Prathab Murugan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.myhome.services.springdatajpa;
+
+import com.myhome.controllers.dto.PaymentDto;
+import com.myhome.controllers.dto.mapper.PaymentMapper;
+import com.myhome.domain.HouseMember;
+import com.myhome.domain.HouseMemberDocument;
+import com.myhome.domain.Payment;
+import com.myhome.repositories.CommunityAdminRepository;
+import com.myhome.repositories.HouseMemberDocumentRepository;
+import com.myhome.repositories.HouseMemberRepository;
+import com.myhome.repositories.PaymentRepository;
+import com.myhome.services.HouseMemberDocumentService;
+import com.myhome.services.PaymentService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * Implements {@link PaymentService} and uses Spring Data JPA Repository to do its work
+ */
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class PaymentSDJpaService implements PaymentService {
+  private final PaymentRepository paymentRepository;
+  private final PaymentMapper paymentMapper;
+  private final CommunityAdminRepository communityAdminRepository;
+  private final HouseMemberRepository houseMemberRepository;
+  private final HouseMemberDocumentService houseMemberDocumentService;
+  private final HouseMemberDocumentRepository houseMemberDocumentRepository;
+
+  @Override
+  public PaymentDto schedulePayment(PaymentDto request) {
+    generatePaymentId(request);
+    return createPaymentInRepository(request);
+  }
+
+  @Override
+  public Set<Payment> listAll() {
+    return listAll(200, 0);
+  }
+
+  @Override
+  public Set<Payment> listAll(Integer limit, Integer start) {
+    return paymentRepository.findAll(PageRequest.of(start, limit)).toSet();
+  }
+
+  @Override
+  public Optional<PaymentDto> getPaymentDetails(String paymentId) {
+    Optional<Payment> payment = paymentRepository.findByPaymentId(paymentId);
```

Comment:
```
return paymentRepository.findByPaymentId(paymentId)
                           .map(paymentMapper::paymentToPaymentDto);
```

Reply:
Shouldn't that be left as findPaymentDetails since it may not have a result?

Reply:
I don't understand you here.
```
paymentRepository.findByPaymentId(paymentId)
```
returns an `Optional` - in case it is not empty map operation will be done otherwise the empty Optional will be returned